projects/aws/eks-a-snow-admin-ami/provisioners/install_docker.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -x
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Install prerequisites
+sudo apt-get update
+sudo apt-get install -y apt-transport-https ca-certificates curl gnupg-agent software-properties-common
+
+# Install docker gpg key
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+
+# Adding docker repo
+LSB_RELEASE=$(lsb_release -cs)
+ARCH=$(dpkg --print-architecture)
+SOURCE="deb [arch=$ARCH] https://download.docker.com/linux/ubuntu $LSB_RELEASE stable"
```

Comment:
Why pull from docker vs using the default package canonical provides? Or is there not a docker in canonicals rpm repo?

Reply:
This are the official instructions to install docker in ubuntu:
https://docs.docker.com/engine/install/ubuntu/

This is also how the official AWS image builder component install docker

I didn't find any canonical rpm but there seems to be a [snap](https://snapcraft.io/install/docker/ubuntu)